hugr-core/src/extension/op_def/serialize_signature_func.rs

```diff
@@ -0,0 +1,171 @@
+use serde::{Deserialize, Serialize};
+
+use super::{CustomValidator, PolyFuncTypeRV, SignatureFunc};
+#[derive(serde::Deserialize, serde::Serialize, PartialEq, Debug, Clone)]
+struct SerSignatureFunc {
+    /// If the type scheme is available explicitly, store it.
+    signature: Option<PolyFuncTypeRV>,
+    /// Whether an associated binary function is expected.
+    /// If `signature` is `None`, a true value here indicates a custom compute function.
+    /// If `signature` is not `None`, a true value here indicates a custom validation function.
+    binary: bool,
+}
+
+pub(super) fn serialize<S>(value: &super::SignatureFunc, serializer: S) -> Result<S::Ok, S::Error>
```

Comment:
can('t) you do a `#[serde(into=....)]` to avoid having to manually implement `fn serialize` ?

Reply:
that requires `Clone` on `SignatureFunc` which I didn't necessarily want to add all the way down.